web/app/js/components/ServiceMesh.jsx

```diff
@@ -152,10 +157,11 @@ export default class ServiceMesh extends React.Component {
       return {
         name: title,
         pods: _.map(matchingPods, p => {
+          let isRunning = parseInt(p.runningPodCount, 10) > 0;
+          let isFailed = parseInt(p.failedPodCount, 10) !== 0;
           return {
             name: p.resource.name,
-            // we need an endpoint to return the k8s status of these pods
-            value: _.size(matchingPods) > 0 ? "good" : "neutral"
+            value: isFailed ? "poor" : isRunning ? "good" : "neutral"
```

Comment:
if you reimplemented `barColor` like this:
```js
const barColor = (runningPodCount, failedPodCount) => {
  if (failedPodCount > 0) {
    return "poor";
  } else if (runningPodCount == 0) {
    return "neutral";
  } else {
    return "good";
  }
};
```

could you then re-use that function here?
```js
value: barColor(parseInt(p.runningPodCount, 10), parseInt(p.failedPodCount, 10));
```

Reply:
Oooh, good idea!